scripts/server.py

```diff
@@ -550,66 +567,544 @@ def handle_client(conn, addr):
+                        if req_status_code in bw_request_status_retry_for:
+                            # Check if we're still table to retry
+                            if submission_retries < bw_request_max_retries:
+                                can_retry = True
+                                submission_retries += 1
+                                request_backoff_time = bw_request_backoff_factor * (2 ** (submission_retries - 1))
+
+                                if debug_birdweather_submissions:
+                                    print(
+                                        f"BirdWeather Submission Error:: Detection POST - HTTP Request Exception for {soundscape_url}, Retrying after {request_backoff_time}s, Retry Attempt:{submission_retries} - {request_exc.response.reason}",
+                                        flush=True)
+                                else:
+                                    print(
+                                        f"BirdWeather Submission Error:: Detection POST - HTTP Request Exception for {soundscape_url}, Retrying after {request_backoff_time}s, Retry Attempt:{submission_retries}",
+                                        flush=True)
+                        else:
+                            print(
+                                f"BirdWeather Submission Error:: Detection POST - HTTP Request Exception for {soundscape_url}, Not retrying - Retries Attempted:{submission_retries} - {request_exc}",
+                                flush=True)
+                    except (
+                            requests.exceptions.JSONDecodeError,
+                            requests.exceptions.InvalidJSONError) as json_error_exec:
+                        print(
+                            f'BirdWeather Submission Error:: Detection POST - Something went wrong decoding JSON data - {json_error_exec}',
+                            flush=True)
+                    except (AttributeError, ValueError, KeyError) as value_exec:
+                        print(
+                            f'BirdWeather Submission Error:: {retry_text}Detection POST - Something went wrong accessing JSON data - {value_exec} not found',
+                            flush=True)
+                    except BaseException as dp_ex:
+                        print(
+                            f'BirdWeather Submission Error:: {retry_text}Detection POST - Something went wrong - {dp_ex}',
+                            flush=True)
+                    finally:
+                        if not can_retry:
+                            # Stop the loop as we can't be sure we can retry on this error
+                            break
+                        else:
+                            # We're retrying, retry after the calculated backoff time
+                            if request_backoff_time > 0:
+                                time.sleep(request_backoff_time)
+                                continue
+                else:
```

Comment:
Can you turn this around maybe? This belongs to an if almost 200 lines ago. 
```
if detection_url is None or detection_post_json is None or soundscape_id is None:
    # handle this
else:
    # The 200 lines above
```

Reply:
I removed both the if statements around section so the cascading else towards has been removed too